alacritty/src/config/ui_config.rs

```diff
@@ -65,7 +66,10 @@ impl Default for UIConfig {
 impl UIConfig {
     #[inline]
     pub fn background_opacity(&self) -> f32 {
-        self.background_opacity.as_f32()
+        match self.background_opacity {
+            Some(x) => x.as_f32(),
+            None => self.window.background_opacity(),
+        }
```

Comment:
This might work:

```rust
self.background_opacity.map(Percentage::as_f32).unwrap_or_else(|| self.window.background_opacity())
```

Reply:
Yes, looks like it would after some mental gymnastics. Not too used to the unwrap-chain bits..  Is that any clearer though?

Reply:
Yeah, I'd recommend this over the existing code.

Reply:
Okay, that's done.